ops/testing.py

```diff
@@ -1559,21 +1590,31 @@ def set_secret_content(self, secret_id: str, content: Dict[str, str]):
     def grant_secret(self, secret_id: str, observer: AppUnitOrName):
         """Grant read access to this secret for the given observer application or unit.
 
+        For user secrets, grant access to the application, simulate the
+        `juju grant-secret` command.
+
         If the given application or unit has already been granted access to
         this secret, do nothing.
 
         Args:
             secret_id: The ID of the secret to grant access to. This should
-                normally be the return value of :meth:`add_model_secret`.
+                normally be the return value of :meth:`add_charm_secret`.
             observer: The name of the application (or specific unit) to grant
                 access to. A relation between this application and the charm
                 under test must already have been created.
         """
         secret = self._ensure_secret(secret_id)
+        app_or_unit_name = _get_app_or_unit_name(observer)
+
+        # user secrets
+        if secret.owner_name == self.model.uuid:
+            secret.user_secrets_grants.add(app_or_unit_name)
```

Comment:
It's true that from Harness's standpoint, there is no need to grant it to the local app but since it's simulating `juju grant-secret`, technically it can grant to other apps. Should I keep it as a set? Or simply as a str of the app name?

Reply:
My point is that Harness doesn't run anything from other apps, so it essentially has no effect. I think keeping it is ok though.